contribs/src/main/java/com/netflix/conductor/contribs/kafka/KafkaPublishTask.java

```diff
@@ -0,0 +1,271 @@
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		if (Objects.isNull(input.getValue())) {
+			task.setReasonForIncompletion(MISSING_KAFKA_BODY);
+			task.setStatus(Task.Status.FAILED);
+			return;
+		}
+
+		try {
+			Future<RecordMetadata> recordMetaDataFuture = kafkaPublish(input);
+			try {
+				recordMetaDataFuture.get();
+				task.setStatus(Task.Status.COMPLETED);
+				long timeTakenToCompleteTask = Instant.now().toEpochMilli() - taskStartMillis;
+				logger.info("Published message {}, Time taken {}", input, timeTakenToCompleteTask);
+
+			} catch (ExecutionException ec) {
+				logger.error("Failed to invoke kafka task - execution exception {}", ec);
+				task.setStatus(Task.Status.FAILED);
+				task.setReasonForIncompletion(FAILED_TO_INVOKE + ec.getMessage());
+			}
+		} catch (Exception e) {
+			logger.error(String.format("Failed to invoke kafka task for input {} - unknown exception: {}", input), e);
+			task.setStatus(Task.Status.FAILED);
+			task.setReasonForIncompletion(FAILED_TO_INVOKE + e.getMessage());
+		}
+	}
+
+	/**
+	 * @param input Kafka Request
+	 * @return Future for execution.
+	 */
+	 private Future<RecordMetadata> kafkaPublish(KafkaPublishTask.Input input) throws Exception {
+
+		long startPublishingEpochMillis = Instant.now().toEpochMilli();
+
+		Producer producer = producerManager.getProducer(input);
+
+		long timeTakenToCreateProducer = Instant.now().toEpochMilli() - startPublishingEpochMillis;
+
+		logger.info("Time taken getting producer {}", timeTakenToCreateProducer);
+
+		Object key = getKey(input);
+
+		Iterable<Header> headers = input.getHeaders().entrySet().stream()
+				.map(header -> new RecordHeader(header.getKey(), String.valueOf(header.getValue()).getBytes()))
+				.collect(Collectors.toList());
+		ProducerRecord rec = new ProducerRecord(input.getTopic(), null,
+				null, key, om.writeValueAsString(input.getValue()), headers);
+
+		Future send = producer.send(rec);
+		producer.close();
+
+		long timeTakenToPublish = Instant.now().toEpochMilli() - startPublishingEpochMillis;
+		logger.info("Time taken publishing {}", timeTakenToPublish);
+
+		return send;
+	}
+
+	@VisibleForTesting
+	Object getKey(Input input) {
+
+		String keySerializer = input.getKeySerializer();
+
+		if (LongSerializer.class.getCanonicalName().equals(keySerializer)) {
+			return Long.parseLong(String.valueOf(input.getKey()));
+		} else if (IntegerSerializer.class.getCanonicalName().equals(keySerializer)) {
+			return Integer.parseInt(String.valueOf(input.getKey()));
+		} else {
+			return String.valueOf(input.getKey());
+		}
+
+	}
+
+	@Override
+	public boolean execute(Workflow workflow, Task task, WorkflowExecutor executor) {
+		return false;
+	}
+
+	@Override
+	public void cancel(Workflow workflow, Task task, WorkflowExecutor executor) {
+		task.setStatus(Task.Status.CANCELED);
+	}
+
+	@Override
+	public boolean isAsync() {
+		return true;
+	}
+
+
+	public static class Input {
```

Comment:
Can you help me understand why this class is inside KafkaPublishTask?

Reply:
This is internal to the Kafka task.  I am following the convention of HTTP task.